### DISCUSSION TITLE
returning error messages about custom options?

### DISCUSSION THREAD

**Discussion by rvandegrift** [pytest-dev/pytest#8609] [Q&A] [answered]
Hello,

My `conftest.py` defines some extra options.  I'd like to validate the input and return detailed error messages if it's wrong, but I can't figure out how to do that.  Here's a silly example:

```python
def wooo(value):
    if value != 'wooo':
        raise ValueError(f'{value} is not equal to wooo')


def pytest_addoption(parser):
    parser.addoption('--wooo', action='store', required=True, type=wooo)
```

When I provide bad input, pytest swallows my error message and replaces it with a generic one:
```sh
$ python3 -m pytest --wooo=asdf
ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
__main__.py: error: argument --wooo: invalid wooo value: 'asdf'
```

Is there any way to get my validation's error returned to the user?

**Comment by The-Compiler** [comment, marked as answer]
That's a feature of the [underlying argparse module](https://docs.python.org/3/library/argparse.html#type).

You could raise [a `pytest.UsageError`](https://docs.pytest.org/en/6.2.x/reference.html#pytest.UsageError) instead, which will instead display:

```
$ pytest  --wooo=asdf
ERROR: asdf is not equal to wooo
```

**Reply by rvandegrift** [reply]
awesome, that works - I've opened https://github.com/pytest-dev/pytest/pull/8610 to add some details on this to the docs.
